src/auth/oauth2client.ts

```diff
@@ -793,6 +804,15 @@ export class OAuth2Client extends AuthClient {
       return {headers: this.addSharedMetadataHeaders(headers)};
     }
 
```

Comment:
The check above:
`if (!thisCreds.access_token && !thisCreds.refresh_token && !this.apiKey) {`

It should be updated to included the `refreshHandler`:
`if (!thisCreds.access_token && !thisCreds.refresh_token && !this.apiKey && !this.refreshHandler) {`